### DISCUSSION TITLE
Proposal for Integrating a Full-Fledged Rich Text Editor

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#5995] [Plugin Suggestions]
Hey Folks,

The current default Rich Text Editor is somewhat functional, but it falls short for production-level use. While we can integrate third-party editors like CKEditor and TinyMCE, the integration process is often complicated.

CKEditor 4 integration is relatively easy, but version 5 and TinyMCE can be more challenging. Therefore, why not have a built-in, powerful Rich Text Editor with the following recommended features? These options are inspired by email editors, but feel free to suggest additional features.

**Rich Text Editor Options:**

1. **Bold** - Make selected text bold.
2. **Italic** - Make selected text italic.
3. **Strikethrough** - Add a strikethrough to the text.
4. **Underline** - Underline the text.
5. **Font Size** - Set font size for specific text.
6. **Font Color** - Set font color for specific text.
7. **Font Background Color** - Set background color for specific text.
8. **Font Family** - Change the font family for specific text.
9. **Remove Formatting** - Remove formatting from the text.
10. **List** - Bullet and numbered list options.
11. **Link** - Basic link styling options like changing link color and removing text decoration.
12. **Personalization** - Add text or emoji at the cursor position, useful for email with shortcodes.

I hope this discussion will get at least 10 votes!

@artf

**Comment by artf** [comment]
We had a [discussion about decoupling RTE](https://github.com/GrapesJS/grapesjs/discussions/5754#discussioncomment-8810479) a time ago and I think the idea of implementing something similar with ProseMirror as a plugin, where also the UI is open for customization, it would be way to go.

**Comment by GoodPHP** [comment]
Froala Inline Text Editor - https://gjs.market/products/froala-inline-text-editor
TinyMCE 6 Inline Text Editor - https://gjs.market/products/tinymce-6-inline-text-editor
CKEditor 5 - https://gjs.market/products/ckeditor-5-inline-text-editor
Kendo - https://gjs.market/products/kendo-rich-text-editor-for-grapesjs
